  public void testCallNextTooEarly() {
    final IteratorChain<String> c = new IteratorChain<>();
    c.addIterator(makeIterator("a",3));
    try {
      c.next();
      fail("Calling next() before hasNext() should throw RuntimeException");
    } catch(RuntimeException asExpected) {
      // we're fine
    }
  }

